tests/test_other.py

```diff
@@ -666,6 +666,19 @@ def test_cmake_compile_features(self):
     emscripten_features = '\n'.join([x for x in emscripten_features.split('\n') if '***' in x])
     self.assertTextDataIdentical(native_features, emscripten_features)
 
+  # Test that concat generator arguments(e.g. "-GNinja" instead of "-G Ninja") can be correctly recognized passed to cmake
+  # Windows Only
+  def test_cmake_with_concat_generator_arg(self):
+    if not utils.which('ninja'):
+      self.skipTest('Skipped concat generator argument test since ninja not found')
+    if not WINDOWS:
+      self.skipTest('Skipped concat generator argument test on non-Windows platforms')
+    # use -Wno-dev to suppress an irrelevant warning about the test files only.
+    cmd = [EMCMAKE, 'cmake', '-GNinja', '-Wno-dev', test_file('cmake/cpp_lib')]
+    env = os.environ.copy()
+    self.run_process(cmd, env=env)
+    self.assertExists(self.get_dir() + '/build.ninja')
```

Comment:
Just checking, does this fail without the change to emcmake.py?

Reply:
Yes, as described by the issue, if there's valid mingw_makefile generator in the env, ninja will not be used.